ember-bootstrap/addon/utils/decorators/arg.ts

```diff
@@ -0,0 +1,33 @@
+import type Component from '@glimmer/component';
+
+type DecoratorPropertyDescriptor = PropertyDescriptor & {
+  initializer?: () => void;
+};
+
+/**
+ * @internal
+ */
+function arg(target: object, key: string): void;
```

Comment:
With generics, is it possible to type args like `<T extends object>(target: T, key: keyof T)`?

Reply:
IIRC I tried that, but at some point I gave up. `key` has to be a keyof this.args actually, see L23. But given this is used as a decorator, which has its own expected signature (with TS actually disagreeing on our implementation), this was not working easily, or I was missing something...

Reply:
I'm not that familiar with creating decorators, so I believe you if you say that it's not working. :sweat_smile: